src/components/protocol_handler/src/service_status_update_handler.cc

```diff
@@ -28,39 +28,42 @@ void ServiceStatusUpdateHandler::OnServiceUpdate(
     const protocol_handler::ServiceType service_type,
     ServiceStatus service_status) {
   using namespace hmi_apis;
+  typedef utils::Optional<Common_ServiceUpdateReason::eType>
+      UpdateReasonOptional;
   auto hmi_service_type = GetHMIServiceType(service_type);
   Common_ServiceEvent::eType service_event;
-  Common_ServiceUpdateReason::eType service_update_reason;
+  UpdateReasonOptional service_update_reason =
+      UpdateReasonOptional(UpdateReasonOptional::EMPTY);
 
   switch (service_status) {
     case ServiceStatus::SERVICE_RECEIVED: {
       service_event = Common_ServiceEvent::REQUEST_RECEIVED;
```

Comment:
@mked-luxoft can you try to use early returns from switch? If no, please move switch to a function